src/identity.rs

```diff
@@ -34,6 +34,23 @@ pub mod mock {
     };
 }
 
+/// Gets the Identity for ztunnel itself using the Downward API
+pub fn get_ztunnel_identity() -> Result<Identity, Error> {
```

Comment:
rather than read this each time we want to fetch it, we should just put it in config.rs and read once at startup. Then invalid config will fail fast instead of later at runtime.